recipes/kissfft/all/conanfile.py

```diff
@@ -0,0 +1,71 @@
+import os
+from conans import ConanFile, CMake, tools
+
+
+class KissfftConan(ConanFile):
+    name = "kissfft"
+    license = "BSD-3-Clause"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/mborgerding/kissfft"
+    description = "a Fast Fourier Transform (FFT) library that tries to Keep it Simple, Stupid"
+    topics = ("conan", "fft", "kiss", "frequency-domain", "fast-fourier-transform")
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False],
+               "fPIC": [True, False],
+               "datatype": ["float", "double", "int16_t", "int32_t", "simd"],
+               "use_alloca": [True, False]}
+    default_options = {"shared": False,
+                       "fPIC": True,
+                       "datatype": "float",
+                       "use_alloca": False}
+    exports_sources = ["CMakeLists.txt"]
+    generators = "cmake",
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        cmake = CMake(self)
+        cmake.definitions["KISSFFT_PKGCONFIG"] = False
+        cmake.definitions["KISSFFT_STATIC"] = not self.options.shared
+        cmake.definitions["KISSFFT_TEST"] = False
+        cmake.definitions["KISSFFT_TOOLS"] = False
+        cmake.definitions["KISSFFT_DATATYPE"] = self.options.datatype
+        cmake.definitions["KISSFFT_USE_ALLOCA"] = self.options.use_alloca
+        cmake.configure()
+        self._cmake = cmake
+        return self._cmake
+
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy("COPYING", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+
+    def package_info(self):
```

Comment:
pkg_config name is missing: https://github.com/mborgerding/kissfft/blob/8f47a67f595a6641c566087bf5277034be64f24d/CMakeLists.txt#L323-L354
CMake imported target is not correct, they add a suffix: https://github.com/mborgerding/kissfft/blob/8f47a67f595a6641c566087bf5277034be64f24d/CMakeLists.txt#L235

Reply:
I am not sure what do you mean, I am not adding CMake imported target there.

Reply:
It doesn't really matter, default imported targets of cmake_find_package[_multi] generators are base on recipe name, which are wrong for this lib.

Reply:
yes, maybe they are. I wasn't checking all the generators, that wasn't the goal, they could be added on demand.

Reply:
`cmake_find_package` and `cmake_find_package_multi` are promoted by conan team, so I believe that an effort should be made to have them correct in the first iteration. And if it requires to much work for a first iteration (many complex components for example), a FIXME should be added so that we can quickly identify those recipes.

Usually it's simple, when you have to delete a cmake or pkgconfig folder in `package()`, it's likely that you'll have to write something in `package_info()` for those generators.

Reply:
yes, maybe it's fair, but I guess it needs more visibility. it's not strictly enforced by the hook right now, and it requires some manual process - probably somehow inspect files which are already deleted (which is tricky). maybe it could be automated or semi-automated, I have no idea right now.